crates/bevy_reflect/src/serde/ser.rs

```diff
@@ -317,8 +317,13 @@ impl<'a> Serialize for EnumSerializer<'a> {
 
         match variant_type {
             VariantType::Unit => {
-                if self.enum_value.reflect_module_path() == Some("core::option")
-                    && self.enum_value.reflect_type_ident() == Some("Option")
+                let table = self
+                    .enum_value
+                    .get_represented_type_info()
+                    .map(|i| i.type_path_table());
```

Comment:
Sorry, what I meant to say was that we fetch the represented type on line 289: https://github.com/bevyengine/bevy/blob/61b83cfc0780cc2b86561cb3f8c7512037f09fca/crates/bevy_reflect/src/serde/ser.rs#L289-L294

So we can use that value instead of doing a second call.
```suggestion
                let table = type_info.map(|i| i.type_path_table());
```